mapper/tedge_mapper/src/sm_c8y_mapper/tests.rs

```diff
@@ -1,81 +1,83 @@
-use crate::sm_c8y_mapper::mapper::CumulocitySoftwareManagement;
-use mqtt_client::Client;
-use mqtt_tests::test_mqtt_server::MqttProcessHandler;
-use mqtt_tests::with_timeout::{Maybe, WithTimeout};
-use serial_test::serial;
-use std::{io::Write, time::Duration};
-use tedge_config::{ConfigRepository, TEdgeConfig, TEdgeConfigLocation};
-use tokio::task::JoinHandle;
-
-const TEST_TIMEOUT_MS: Duration = Duration::from_millis(1000);
-
-#[tokio::test]
-#[serial]
-async fn mapper_publishes_a_software_list_request() {
-    // The test assures the mapper publishes request for software list on `tedge/commands/req/software/list`.
-    let broker = mqtt_tests::test_mqtt_broker();
-
-    let mut messages = broker
-        .messages_published_on("tedge/commands/req/software/list")
-        .await;
-
-    // Start the SM Mapper
-    let sm_mapper = start_sm_mapper(broker.port).await;
-
-    // Expect on `tedge/commands/req/software/list` a software list request.
-    let msg = messages
-        .recv()
-        .with_timeout(TEST_TIMEOUT_MS)
-        .await
-        .expect_or("No message received after a second.");
-    dbg!(&msg);
-    assert!(&msg.contains(r#"{"id":"#));
-
-    sm_mapper.unwrap().abort();
-}
-
-#[tokio::test]
-#[serial]
-async fn mapper_publishes_a_supported_operation_and_a_pending_operations_onto_c8y_topic() {
-    // The test assures the mapper publishes smartrest messages 114 and 500 on `c8y/s/us` which shall be send over to the cloud if bridge connection exists.
-    let broker = mqtt_tests::test_mqtt_broker();
-    let mut messages = broker.messages_published_on("c8y/s/us").await;
-
-    // Start SM Mapper
-    let sm_mapper = start_sm_mapper(broker.port).await;
-
-    // Expect both 114 and 500 messages has been received on `c8y/s/us`, if no msg received for the timeout the test fails.
-    mqtt_tests::assert_received(
-        &mut messages,
-        TEST_TIMEOUT_MS,
-        vec![
-            "114,c8y_SoftwareUpdate,c8y_LogfileRequest\n",
-            "118,software-management\n",
-            "500\n",
-        ],
-    )
-    .await;
-    sm_mapper.unwrap().abort();
-}
-
-#[tokio::test]
-#[serial]
-async fn mapper_publishes_software_update_request() {
-    // The test assures SM Mapper correctly receives software update request smartrest message on `c8y/s/ds`
-    // and converts it to thin-edge json message published on `tedge/commands/req/software/update`.
-    let broker = mqtt_tests::test_mqtt_broker();
-    let mut messages = broker
-        .messages_published_on("tedge/commands/req/software/update")
+#[cfg(test)]
```

Comment:
The change of this file is just to add `#[cfg(test)]`. In terms of functionalities, this file is unrelated to child device support.

Reply:
This is not required as already stated in `mod.rs`.

This is even a bit heavy, since now the module name is `tests::tests` and tagged twice as a test module.